### DISCUSSION TITLE
Integrate as widget into site

### DISCUSSION THREAD

**Discussion by arvindpdmn** [ken107/read-aloud#223] [General]
Rather than asking users to install browser plugin, has anyone used this code as a widget in their site? Any server-side support is needed for this?

**Comment by ken107** [comment]
Hello, you probably looking for https://readaloud.app?tab=embed-website

**Comment by arvindpdmn** [comment]
- Thanks for the link. However, I was wondering if it can be hosted from my own server. The link you shared makes calls to `https://assets.sitespeaker.link`
- Is there a way to customize the styling? I deleted `data-skin="https://assets.sitespeaker.link/embed/skins/default"` but this causes player not to load properly.
- I noticed that when content on the web page changes, as is common on Wikipedia pages or any collaborative site, SiteSpeaker keeps sending the audio file of the old content even when new content is passed via the API. This is also the case when I change the voice. The audio file has the voice I selected the first time for that page. I would treat these as bugs.

Needed by https://github.com/DevopediaOrg/webapp/issues/507
